### PR TITLE
[#1124]  Read only permissions

### PR DESCRIPTION
**Pull request by muloem**
* Implement UI changes regarding read only permissions

### CODE REVIEW

File: Dashboard/app/js/lib/controllers/permissions.js [hunk 1]
```diff
@@ -137,7 +137,56 @@ FLOW.permControl = Ember.Controller.create({
     this.perms.forEach(function (item) {
       //this.set(item.perm,item.value);
     });
-  }
+  },
+
+  /* Given an entity, process the permissions settings for the current user
+    and return the permissions associated with that entity.  Entity is an Ember object*/
+  permissions: function(entity) {
+    var keyId, ancestorIds, permissions = [], currentUserPermissions = FLOW.userControl.currentUserPathPermissions();
```
Comment (jonase) [inline Dashboard/app/js/lib/controllers/permissions.js]: @muloem For my own understanding.. Does `FLOW.userControl.currentUserPathPermissions()` return a map from `id => list of permissions`? For example
```
{
    "123": ["FORM_CREATE", "FORM_READ"],
    "432": ["DATA_CLEANING", "DATA_DELETE"]
}
```

File: Dashboard/app/js/lib/controllers/permissions.js [hunk 2]
```diff
@@ -137,7 +137,56 @@ FLOW.permControl = Ember.Controller.create({
...
+    // check ancestor permissions
+    ancestorIds = entity.get('ancestorIds');
+    if (!ancestorIds) {
+      return permissions;
+    }
+
+    var i;
+    for(i = 0; i < ancestorIds.length; i++){
+      if (ancestorIds[i] in currentUserPermissions) {
+        permissions = permissions.concat(currentUserPermissions[ancestorIds[i]]);
```
Comment (jonase) [inline Dashboard/app/js/lib/controllers/permissions.js]: Should we make sure that `permissions` does not contain duplicates?

File: Dashboard/app/js/lib/controllers/permissions.js (hunk 1 above)
Comment (muloem) [inline Dashboard/app/js/lib/controllers/permissions.js]: yes exactly.

File: Dashboard/app/js/lib/controllers/permissions.js (hunk 2 above)
Comment (muloem) [inline Dashboard/app/js/lib/controllers/permissions.js]: thats a good point. I'll add that change

**Comment by jonase** [comment]
:+1:

**Pull request merged**